rasa/utils/tensorflow/models.py

```diff
@@ -820,11 +825,28 @@ def _get_mask_for(
     def _get_sequence_feature_lengths(
         self, tf_batch_data: Dict[Text, Dict[Text, List[tf.Tensor]]], key: Text
     ) -> tf.Tensor:
-        """Fetches the sequence lengths of real tokens per input example.
+        """Tries to create sequence length information for the given key.
+
+        Note that, if no explicit sequence length information is present in the batch,
+        this function assumes the respective sequence-level features are all empty.
+        This behaviour useful in case we deal with empty sequences only (e.g.
+        DIET without transformer layers and no need for sequential output).
```

Comment:
```suggestion
        DIET without transformer layers and hence no need for sequential input).
```

Reply:
I do mean output here. Assume we have no transformer layers. In that strange case where you do entity recognition, you still need sequential output (and yes hence, the sequential input) sequence. But if we don't require sequential output, then we also won't make use of the sequential input (cause that won't end up anywhere).

Reply:
> Assume we have no transformer layers. In that strange case where you do entity recognition

Do we support entity recognition in that case? 🙈  Looks like we should ideally raise an exception and suggest adding transformer layers.

Reply:
I think so, nothing restricts that yet... Baseline that's not using context? 😄  We also could emit a warning?

Reply:
A warning sounds good but let's add it in a separate PR. I think we've had discussions related to this topic somewhere above. Can add it to a related issue.